### PR TITLE
Port network to tokio 0.2 and tonic

### PR DESCRIPTION
**Pull request by mzabaluev**

### CODE REVIEW

**Comment by eugene-babichenko** [comment]
@mzabaluev pushed all made in #1940

Can you take a look please?

File: jormungandr/src/network/p2p/comm.rs [hunk 1]
```diff
@@ -611,40 +607,38 @@ impl Peers {
...
-                    );
-                    entry.remove();
-                    e.into_item()
-                })
-            } else {
-                Err(gossip)
-            }
-        })
+        self.inner()
+            .and_then(move |mut map| {
```
Comment (NicolasDP) [inline jormungandr/src/network/p2p/comm.rs:618]: can't we do the other way around:

```rust
   let mut map = self.inner().await?;
   if let Some(mut entry) = map.entry(target) {
      ...
   }
```

Comment (NicolasDP) [reply jormungandr/src/network/p2p/comm.rs:618]: I just find it is easier to read (separate the async logic from the processing logic

File: jormungandr/src/network/p2p/comm.rs [hunk 2]
```diff
@@ -611,40 +607,38 @@ impl Peers {
...
+                        e.into_item()
+                    })
+                } else {
+                    Err(gossip)
+                }
+            })
+            .await
     }
 
     pub fn refresh_peer_on_block<E>(&self, node_id: Id) -> impl Future<Item = bool, Error = E> {
```
Comment (NicolasDP) [inline jormungandr/src/network/p2p/comm.rs:644]: do we still need these old futures?

File: jormungandr/src/network/p2p/comm.rs (hunk 1 above)
Comment (mzabaluev) [reply jormungandr/src/network/p2p/comm.rs]: You are right, changed in another commit.

File: jormungandr/src/network/p2p/comm.rs (hunk 2 above)
Comment (mzabaluev) [reply jormungandr/src/network/p2p/comm.rs]: I have converted these in another commit.

**Comment by NicolasDP** [comment]
Woohoo!

**Pull request merged**